### ISSUE TITLE
`__VERIFIER_nondet_memory` support

### ISSUE THREAD

**Issue by sim642** [goblint/analyzer#1846]
SV-COMP 2026 defines a new function:

>  `__VERIFIER_nondet_memory(void *, size_t)`: This function initializes the given memory block with arbitrary values. The first argument must be a valid pointer to the start of a memory block of the given size. The second argument specifies the size of the memory to initialize and must match the size of the memory block that the first argument points to. The dereference of any pointer value set through this method results in undefined behavior. This means that pointer values must be explicitly set through different means before they can be dereferenced. The verification tool can assume that `__VERIFIER_nondet_memory` is implemented as follows:
> ```c
> void __VERIFIER_nondet_memory(void *mem, size_t size) {
>     unsigned char *p = mem;
>     for (size_t i = 0; i < size; i++) {
>         p[i] = __VERIFIER_nondet_uchar();
>     }
> }
> ```   
> 
> Example uses of `__VERIFIER_nondet_memory`:
> ```c
> struct structType s;
> __VERIFIER_nondet_memory(&s, sizeof(s));
> 
> int * values = malloc(sizeof(int) * 20);
> __VERIFIER_nondet_memory(values, sizeof(int) * 20);
> ```

(https://sv-comp.sosy-lab.org/2026/rules.php)

**Comment by michael-schwarz** [comment]
Interesting! We should probably add tests to SV-COMP right away that do things such as:

~~~C
enum A {
  a, b, c
};

enum A* ptr = malloc(sizeof(enum A));
__VERIFIER_nondet_memory(ptr, sizeof(enum A));


if(*ptr != a && *ptr != b && *ptr != c)) {
   reach_error(); // Reachable (compiler needs to assign at least 2 bits for type enum A)
}
~~~ 

Just to make sure no one has any ideas about this being semantic in any way...
